oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/type/SelectedRecord.java

```diff
@@ -38,7 +38,7 @@ public class SelectedRecord {
     /**
      * Usually an integer value as this is metrics.
      */
-    private String value;
+    private long value;
```

Comment:
Is this safe to change? I can see the protocol was set as String to represent an int/long.

Reply:
> Is this safe to change? I can see the protocol was set as String to represent an int/long.

This must be changed, we have to re-calculate the metrics value when aggregating from multiple tables, and have to cast this to `String` again when sending to protocol

Reply:
Isn't this logic staying inside JDBC and/or Sharding-MySQL DAOs? I think ElasticSearch and BanyanDB don't need this process.

Reply:
Not limited to mysql ShardingSphere. All JDBC implementations need this.

Reply:
Yes. But changing this means all storage layers. I know that is no side effect, as we are using int/long as always.
I am more about wondering, SelectedRecord is already a GraphQL object, how to convert it?

Reply:
> Yes. But changing this means all storage layers. I know that is no side effect, as we are using int/long as always.
I am more about wondering, SelectedRecord is already a GraphQL object, how to convert it?

`SelectedRecord` doesn't look to be a GraphQL object, correct me if I am wrong. It is always converted to `TopNRecord`, `TopNRecord` is a GraphQL object

Reply:
Read this, 

- https://github.com/apache/skywalking-query-protocol/blob/master/metrics-v2.graphqls#L160
- https://github.com/apache/skywalking-query-protocol/blob/master/metrics-v2.graphqls#L133

Reply:
> Yes. But changing this means all storage layers. I know that is no side effect, as we are using int/long as always.
> I am more about wondering, SelectedRecord is already a GraphQL object, how to convert it?

I can query the SelectedRecord correctly, even if I change the String type to long, do you want me to change the protocol type in query protocol repo still, I can do that if you want?

Reply:
I don't want to change the query protocol, in case of any unexpected case. Let's keep String in the final response.